bridges/snowbridge/pallets/system/src/lib.rs

```diff
@@ -574,6 +594,61 @@ pub mod pallet {
 			});
 			Ok(())
 		}
+
+		/// Sends a message to the Gateway contract to register a new
+		/// token that represents `asset`.
+		///
+		/// - `origin`: Must be `MultiLocation` from sibling parachain
+		#[pallet::call_index(10)]
+		#[pallet::weight(T::WeightInfo::register_token())]
+		pub fn register_token(
+			origin: OriginFor<T>,
+			asset: Box<VersionedLocation>,
+			metadata: AssetRegistrarMetadata,
+		) -> DispatchResult {
+			let origin_location: Location = T::SiblingOrigin::ensure_origin(origin)?;
```

Comment:
I think its safe to make token registration permissionless, allowing anyone to register PNA tokens, as long as they pay upfront fees like other permissionless calls such as `CreateAgent`.

Reply:
https://github.com/Snowfork/polkadot-sdk/pull/128/commits/0f0fd8e88e3490a912e287025676ace94d6f0e03